prow/create-secrets-for-workload-cluster.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+# Source development/pre-install-prow.sh
+
+set -o errexit
+
+if [ -z "$PROJECT" ]; then
+      echo "\$PROJECT is empty"
+      exit 1
+fi
+
+if [ -z "$GCLOUD_COMPUTE_ZONE" ]; then
+      echo "\$GCLOUD_COMPUTE_ZONE is empty"
+      exit 1
+fi
+
+if [ -z "$WORKLOAD_CLUSTER_NAME" ]; then
+      echo "\$WORKLOAD_CLUSTER_NAME is empty"
+      exit 1
+fi
+
+if [ -z "$GOPATH" ]; then
+      echo "\$GOPATH is not set"
+      exit 1
+fi
+
+### Cloning k8s.io/test-infra is a prerequisite here
+git clone "git@github.com:kubernetes/test-infra.git" "${GOPATH}/src/k8s.io/test-infra" || cd "${GOPATH}/src/k8s.io/test-infra/prow"
```

Comment:
It is very dangerous because you are cloning master branch from test-infra, and at any time they can introduce breaking changes that will cause failure of our script. You should use specific commit ID, ideally the same which is used in starter.yaml

Reply:
Added the git SHA hard reset after git clone